### ISSUE TITLE
Enable @typescript-eslint/switch-exhaustiveness-check rule

### ISSUE THREAD

**Issue by Scrub1492** [themoeway/yomitan#449]
Enable @typescript-eslint/switch-exhaustiveness-check rule, as discussed in https://github.com/themoeway/yomitan/pull/431#discussion_r1435927260. 

I tried turning it on, but it started yelling about requiring "parserOptions.project" property for @typescript-eslint/parser. Maybe @toasted-nutbread will have a better idea about how to fix this.

**Comment by toasted-nutbread** [comment]
Yes I encountered this also, not sure why it doesn't work given that the project field is filled out. I've been trying to do a few upgrades to some dependencies to see if that helps but have encountered other issues in the process.
